tutorials/nameservice/base/x/auction/Makefile

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/make -f
+
+DOCKER := $(shell which docker)
+
+##################
+###   Build   ####
+##################
+
+test:
+	@echo "--> Running tests"
+	go test -v ./...
+
+test-integration:
+	@echo "--> Running integration tests"
+	cd integration; go test -v ./...
+
+.PHONY: test test-integration
+
+###################
+###  Protobuf  ####
+###################
+
+protoVer=0.13.2
+protoImageName=ghcr.io/cosmos/proto-builder:$(protoVer)
+protoImage=$(DOCKER) run --rm -v $(CURDIR):/workspace --workdir /workspace $(protoImageName)
+
+proto-all: proto-format proto-lint proto-gen
+
+proto-gen:
+	@echo "Generating protobuf files..."
+	@$(protoImage) sh ./scripts/protocgen.sh
+	@go mod tidy
+
+proto-format:
+	@$(protoImage) find ./ -name "*.proto" -exec clang-format -i {} \;
+
+proto-lint:
+	@$(protoImage) buf lint
+
+.PHONY: proto-all proto-gen proto-format proto-lintll proto-gen proto-gen-any proto-swagger-gen proto-format proto-lint proto-check-breaking proto-update-deps mocks
+
+##################
+###  Linting  ####
+##################
+
+golangci_lint_cmd=golangci-lint
+golangci_version=v1.51.2
+
+lint:
+	@echo "--> Running linter"
+	@go install github.com/golangci/golangci-lint/cmd/golangci-lint@$(golangci_version)
+	@$(golangci_lint_cmd) run ./... --timeout 15m
+
+lint-fix:
+	@echo "--> Running linter and fixing issues"
+	@go install github.com/golangci/golangci-lint/cmd/golangci-lint@$(golangci_version)
+	@$(golangci_lint_cmd) run ./... --fix --timeout 15m
+
+.PHONY: lint lint-fix
```

Comment:
is this to show the user?

Reply:
For the dev that is following the tutorial but not end user